File: app/src/main/java/com/drdisagree/iconify/utils/overlay/manager/resource/ResourceEntry.java
package com.drdisagree.iconify.utils.overlay.manager.resource;

@SuppressWarnings({"unused"})
public class ResourceEntry {

    private String packageName;
    private String startEndTag;
    private String resourceName;
    private String resourceValue;

    public ResourceEntry(String packageName, String startEndTag, String resourceName, String resourceValue) {
        this.packageName = packageName;
        this.startEndTag = startEndTag;
        this.resourceName = resourceName;
        this.resourceValue = resourceValue;
    }

    public String getPackageName() {
        return packageName;
    }

    public void setPackageName(String packageName) {
        this.packageName = packageName;
    }

    public String getStartEndTag() {
        return startEndTag;
    }

    public void setStartEndTag(String startEndTag) {
        this.startEndTag = startEndTag;
    }

    public String getResourceName() {
        return resourceName;
    }

    public void setResourceName(String resourceName) {
        this.resourceName = resourceName;
    }

    public String getResourceValue() {
        return resourceValue;
    }

    public void setResourceValue(String resourceValue) {
        this.resourceValue = resourceValue;
    }
}

File: app/src/main/java/com/drdisagree/iconify/utils/overlay/manager/resource/ResourceEntry.kt
package com.drdisagree.iconify.utils.overlay.manager.resource

@Suppress("unused")
class ResourceEntry {

    var packageName: String
    var startEndTag: String
    var resourceName: String
    var resourceValue: String
    private var isPortrait: Boolean
    var isNightMode: Boolean
    private var isLandscape: Boolean

    constructor(
        packageName: String,
        startEndTag: String,
        resourceName: String,
        resourceValue: String
    ) {
        this.packageName = packageName
        this.startEndTag = startEndTag
        this.resourceName = resourceName
        this.resourceValue = resourceValue

        isPortrait = true
        isNightMode = false
        isLandscape = false
    }

    constructor(packageName: String, startEndTag: String, resourceName: String) {
        this.packageName = packageName
        this.startEndTag = startEndTag
        this.resourceName = resourceName

        resourceValue = ""
        isPortrait = true
        isNightMode = false
        isLandscape = false
    }

    fun isPortrait(): Boolean {
        return isPortrait
    }

    fun setPortrait(portrait: Boolean) {
        isPortrait = portrait
        isLandscape = !portrait
    }

    fun isLandscape(): Boolean {
        return isLandscape
    }

    fun setLandscape(landscape: Boolean) {
        isLandscape = landscape
        isPortrait = !landscape
    }
}